### PR TITLE
Fix incorrect current_cumulative_gas_used in end_tx_gadget

### PR DESCRIPTION
**Pull request by lispc**
when block contains >=3 txs. And update tests

### CODE REVIEW

**Comment by lispc** [comment]
@DreamWuGit can you have a review on this?

File: zkevm-circuits/src/evm_circuit/execution/end_tx.rs [hunk 1]
```diff
@@ -256,7 +256,7 @@ impl<F: Field> ExecutionGadget<F> for EndTxGadget<F> {
         } else {
             let rw = &block.rws[(
                 RwTableTag::TxReceipt,
-                (tx.id - 1) * TxReceiptFieldTag::COUNT - 1,
+                (tx.id - 1) * (TxReceiptFieldTag::COUNT + 1) - 2,
```
Comment (DreamWuGit) [inline zkevm-circuits/src/evm_circuit/execution/end_tx.rs:259]: should be: 
` let idx = (tx.id - 2) * (TxReceiptFieldTag::COUNT + 1) + 2;` as earlier discussion ?

File: zkevm-circuits/src/evm_circuit/execution/end_tx.rs [hunk 2]
```diff
@@ -254,9 +254,11 @@ impl<F: Field> ExecutionGadget<F> for EndTxGadget<F> {
         let current_cumulative_gas_used: u64 = if tx.id == 1 {
             0
         } else {
+            // first receipt only needs TxReceiptFieldTag::COUNT(3) rws
+            // later receipts need 4(with one extra cumulative gas read) rws
```
Comment (icemelon) [inline zkevm-circuits/src/evm_circuit/execution/end_tx.rs:258]:
```suggestion
            // first transaction needs TxReceiptFieldTag::COUNT(3) lookups to tx receipt,
            // while later transactions need 4 (with one extra cumulative gas read) lookups
```

**Review by icemelon — APPROVED** [review]
LGTM. one minor comment

**Review by DreamWuGit — APPROVED** [review]
LGTM ! thanks for the fix !

**Pull request merged**